File: routes/admin/catedel.js

var res = require('./lib/resp.js');
var cates = require("./models/cates.js");
var dbUtils = require('./models/dbUtils.js');


if (!session.get("isAdmin")) {
  return res.redirect('/routes/admin/login.js');
}

var oid = request.formValue('oid');
cates.delByOid(oid);
res.redirect('./catelist.js');

File: routes/admin/catedel.go
package admin

import (
	"github.com/zengming00/go-testShop/framework"
)

func CateDel(ctx *framework.HandlerContext) {
	var isAdmin = false
	if v, ok := ctx.GetSessionVal("isAdmin"); ok {
		isAdmin = v.(bool)
	}
	if !isAdmin {
		ctx.Redirect("/routes/admin/login.go")
	}

	var oid = ctx.R.FormValue("oid")
	ctx.Cates.DelByOid(oid)
	ctx.Redirect("./catelist.go")
}
